--- /usr/lib/qt5/qml/Sailfish/Weather/WeatherBannerForecast.qml
+++ /usr/lib/qt5/qml/Sailfish/Weather/WeatherBannerForecast.qml
@@ -1,6 +1,7 @@
 import QtQuick 2.0
 import Sailfish.Silica 1.0
 import Sailfish.Weather 1.0
+import org.nemomobile.lipstick 0.1
 
 Column {
     id: root
@@ -14,6 +15,8 @@ Column {
                                : (forecastModel.status == Weather.Loading && forecastModel.count > 0) ? 0.4 : 0.0
     Behavior on dataOpacity { FadeAnimation { property: "dataOpacity" } }
 
+    readonly property bool rotated: Lipstick.compositor.topmostWindowOrientation & (Qt.LandscapeOrientation | Qt.InvertedLandscapeOrientation)
+
     function attemptReload() {
         forecastModel.attemptReload()
     }
@@ -84,7 +87,7 @@ Column {
                 timestamp: weatherModel.timestamp
             }
             delegate: Item {
-                width: weatherForecastList.width/5
+                width: weatherForecastList.width / Math.round(5 * (rotated ? Screen.height / Screen.width : 1))
                 height: weatherForecastList.height
                 WeatherForecastItem {
                     highlighted: root.highlighted
