--- /usr/lib/qt5/qml/Sailfish/Weather/WeatherBannerForecast.qml
+++ /usr/lib/qt5/qml/Sailfish/Weather/WeatherBannerForecast.qml
@@ -75,7 +75,7 @@ SilicaListView {
             id: weatherForecastList
 
             anchors.fill: parent
-            clip: true // limit to 5-7 day forecast
+            clip: true
             currentIndex: -1
             interactive: false
             orientation: ListView.Horizontal
@@ -84,7 +87,7 @@ Column {
             delegate: Item {
 
-                // Show 5 items on normal phones, 7 on larger screens
-                readonly property int columnCount: Math.round(weatherForecastList.width / (540 * Theme.pixelRatio/5)) >= 7 ? 7 : 5
+                // Show 5 items on normal phones, proportionally more on larger screens
+                readonly property int columnCount: Math.round(weatherForecastList.width / (540 * Theme.pixelRatio/5))
 
                 width: Math.round(weatherForecastList.width/columnCount)
                 height: weatherForecastList.height
